### PR TITLE
Move namespace handler to service/frontend

### PR DESCRIPTION
**Pull request by rodrigozhou**
**What changed?**
Moved `namespace/handler.go` to `service/frontend/namespace_handler.go`.
Renamed `namespace.ArchivalState` to `namespace.ArchivalConfigState`.
Renamed `namespace.ArchivalEvent` to `namespace.ArchivalConfigEvent`.

**Why?**
`namespace` module is depending on `archiver` module, which will cause potential circular dependencies (eg: `searchattribute` module cannot import `namespace`).

Since the namespace handler is only used by `frontend`, I'm moving it there.
As for the renaming, there's another type name `ArchivalState` already defined in `archiver`, so I'm changing the one in namespace to better distinguish them.

**How did you test it?**
Existing tests.

**Potential risks**
No.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/namespace/archivalConfigStateMachine.go [hunk 1]
```diff
@@ -60,31 +60,31 @@ var (
...
+func (s *ArchivalConfigState) validate() error {
 	if s.State == enumspb.ARCHIVAL_STATE_ENABLED && len(s.URI) == 0 {
 		return errInvalidState
 	}
 	return nil
 }
 
-func (s *ArchivalState) getNextState(
-	e *ArchivalEvent,
+func (s *ArchivalConfigState) GetNextState(
```
Comment (alexshtin) [inline common/namespace/archivalConfigStateMachine.go:84]: OMG, 58!

File: service/frontend/adminHandler.go [hunk 2]
```diff
@@ -174,17 +173,6 @@ func NewAdminHandler(
 		status:                common.DaemonStatusInitialized,
 		numberOfHistoryShards: args.PersistenceConfig.NumHistoryShards,
 		config:                args.Config,
-		namespaceHandler: namespace.NewHandler(
```
Comment (alexshtin) [inline service/frontend/adminHandler.go:177]: :+1:

**Review by alexshtin — APPROVED** [review]

File: service/frontend/namespace_handler.go [hunk 3]
```diff
@@ -98,28 +99,35 @@ const (
...
 	clusterMetadata cluster.Metadata,
-	namespaceReplicator Replicator,
+	namespaceReplicator namespace.Replicator,
 	archivalMetadata archiver.ArchivalMetadata,
 	archiverProvider provider.ArchiverProvider,
 	supportsSchedules dynamicconfig.BoolPropertyFnWithNamespaceFilter,
 	timeSource clock.TimeSource,
-) *HandlerImpl {
-	return &HandlerImpl{
+) *namespaceHandlerImpl {
```
Comment (alexshtin) [inline service/frontend/namespace_handler.go:123]: I think you need to fix this, and add `//nolint` comment for all the rest.

Comment (rodrigozhou) [reply service/frontend/namespace_handler.go:123]: Fixed this one, but gonna leave the others. This diff is a refactor, so I don't wanna silence a lint warning that might be useful later.

**Pull request merged**